lib/ansible/module_utils/ismount.py

```diff
@@ -49,42 +49,71 @@
 # Agreement.
 
 import os
+import sys
 
 
 def ismount(path):
-    """Test whether a path is a mount point
-    clone of os.path.ismount (from cpython Lib/posixpath.py)
-    fixed to solve https://github.com/ansible/ansible-modules-core/issues/2186
-    and workaround non-fixed http://bugs.python.org/issue2466
-    this should be rewritten as soon as python issue 2466 is fixed
-    probably check for python version and use os.path.ismount if fixed
+    """python issue 2466 is fixed then ismount() was rewritten"""
```

Comment:
I would leave a note in here about why we still have this if the function has been fixed upstream.
```suggestion
    """Test whether a path is a mount point
    This is a copy of the upstream version of ismount(). Originally this was copied here as a workaround
    until Python issue 2466 was fixed. Now it is here so this will work on older versions of Python
    that may not have the upstream fix.
    https://github.com/ansible/ansible-modules-core/issues/2186
    http://bugs.python.org/issue2466
    """
```